src/Microsoft.Azure.WebJobs.Host/Bindings/TraceWriter/TraceWriterBinding.cs

```diff
@@ -45,11 +49,21 @@ public Task<IValueProvider> BindAsync(BindingContext context)
                 throw new ArgumentNullException("context");
             }
 
+            TraceWriter trace = context.Trace;
             object tracer = null;
             if (_parameter.ParameterType == typeof(TraceWriter))
             {
-                // bind directly to the context TraceWriter
-                tracer = context.Trace;
+                // If logger functionality is enabled, wrap an ILogger
+                if (_loggerFactory != null)
+                {
+                    ILogger logger = _loggerFactory.CreateLogger(LoggingCategories.Function);
+                    tracer = new CompositeTraceWriter(new[] { trace, new LoggerTraceWriter(context.Trace.Level, logger) }, null, context.Trace.Level);
+                }
+                else
+                {
+                    // bind directly to the context TraceWriter
+                    tracer = trace;
+                }
             }
             else
```

Comment:
The else case handles bindings to TextWriter which most of our customers still use. Don't we want those logs to also go to AppInsights?